TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/ArmSystem.java

```diff
@@ -298,150 +196,56 @@ private void toggleGripper() {
-            } catch (Exception e) {
-
-            }
-
-        }
-        slider.setPower(0);
-        //going down
-        for( int i = 0; i < 1111; i++) {
-            // go up
-            slider.setDirection(Direction.motorDirection(Direction.DOWN));
-            slider.setPower(0.2);
-            if(!limitSwitch.getState()) {
-                break;
-            }
-            try {
-                Thread.sleep(2);
-            } catch (Exception e) {
-
-            }
-        }
-        calibrated = true;
-        this.calibrationDistance = slider.getCurrentPosition();
-        slider.setPower(0);
-        /*
-        // At rest, pre-int, this will be false, i.e. the limit switch IS pressed.
-        if (limitSwitch.getState()) {
-            // Limit switch has been released, cord is under tension.
-            // With cord under tension, lower until switch is depressed.
-            direction = Direction.DOWN;
-        }
-
-        Log.d(TAG,"Calibrating");
-        Log.d(TAG, "Direction: " + direction);
-        Log.d(TAG, "Switch State: " + limitSwitch.getState());
-        // If we're going down and we hit the switch, then we're done
-        if (direction == Direction.DOWN && !limitSwitch.getState()) {
-            slider.setPower(0);
-            calibrated = true;
-            calibrationDistance = slider.getCurrentPosition();
-            slider.setDirection(Direction.motorDirection(Direction.UP));
-            setSliderHeight(0);
-        }
-
-         */
+    private void placeStone () {
+        openGripper();
+        setSliderHeight(getSliderPos() + 1);
+        movePresetPosition(Position.POSITION_HOME);
+        moveHome();
     }
 
-    private void stop() {
-        slider.setPower(0);
+    private void movePresetPosition (Position pos){
+        double[] posArray = pos.getPos();
+        pivot.setPosition(posArray[0]);
+        elbow.setPosition(posArray[1]);
+        wrist.setPosition(posArray[2]);
     }
 
     // Pos should be the # of blocks high it should be
-    private void setSliderHeight(int pos) {
+    public void setSliderHeight ( int pos){
         targetHeight = pos;
+        if (pos < 0) targetHeight = 0;
+        if (pos > 9) targetHeight = 9;
         slider.setTargetPosition(calculateHeight(targetHeight));
         slider.setDirection(Direction.motorDirection(Direction.UP));
         slider.setMode(DcMotor.RunMode.RUN_TO_POSITION);
-        Log.d(TAG, "Set target height to" + calculateHeight(targetHeight));
+        updateHeight(1);
     }
 
     // Little helper method for setSliderHeight
-    private int calculateHeight(int pos) {
+    private int calculateHeight ( int pos){
+        if (pos == 0) return calibrationDistance;
         return START_HEIGHT + calibrationDistance + (pos * INCREMENT_HEIGHT);
     }
 
-    // Should be called every loop
-    private void updateHeight(double speed) {
+    // Must be called every loop
+    public void updateHeight ( double speed){
```

Comment:
Can we rename this to `goToHeight` or `raise` or something? It's not really _updating_ the height...

Reply:
Done